A/Arrow/build_tarballs.jl

```diff
@@ -0,0 +1,39 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "Arrow"
+version = v"0.1.0"
+
+# Collection of sources required to complete build
+sources = [
+    GitSource("https://github.com/apache/arrow.git", "fcd44e4d5e9aea8b540d28e5fb6fc2427d804073")
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir
+cd arrow/cpp/
+mkdir build
+cd build
+cmake -DCMAKE_INSTALL_PREFIX=$prefix -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} -DCMAKE_BUILD_TYPE=Release ..
+make
+make install
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = supported_platforms()
+
+
+# The products that we will ensure are always built
+products = [
+    LibraryProduct("libarrow", :libarrow)
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = Dependency[
+]
+
+# Build the tarballs, and possibly a `build.jl` as well.
+build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; preferred_gcc_version = v"9.1.0")
```

Comment:
Why such a recent version of GCC?  That's unlikely to work with the official binaries of Julia

Reply:
That's actually exactly the reason why the library can't be `dlopen`ed: https://dev.azure.com/JuliaPackaging/Yggdrasil/_build/results?buildId=5588&view=logs&j=01859c06-840e-59ce-0724-571b2b074689&t=fc66dd4d-34e4-5ce9-a2f5-450f663f0150&l=1046
```
/usr/local/bin/../lib/julia/libstdc++.so.6: version `GLIBCXX_3.4.26' not found (required by /agent/_work/2/s/A/Arrow/build/x86_64-linux-gnu/tSLp0J6a/destdir/lib/libarrow.so.100.1.0)
```
#918 uses the default GCC version, v4.8.  I don't see why we should use GCC v9 here